addons/auth/addon/authenticators/base.js

```diff
@@ -101,15 +106,41 @@ export default class BaseAuthenticator extends SimpleAuthBaseAuthenticator {
    * @param {string} username
    * @return {object}
    */
-  normalizeData(data, username) {
+  async normalizeData(data) {
     // Pull fields up from `data.attributes` for easier access in JavaScript.
     // The `attributes` field exists on the Go side for its convenience but is
     // unnecessary here.
     Object.assign(data, data.attributes);
     // Add booleans indicated the scope type
     data.isGlobal = data?.scope?.type === 'global';
     data.isOrg = data?.scope?.type === 'org';
-    if (username) data.username = username;
+
+    try {
+      const adapter = this.store.adapterFor('application');
+      const findAccountRecordURL = adapter.buildURL(
+        'account',
+        data?.account_id,
+        {},
+        'findRecord',
+      );
+      const response = await waitForPromise(
+        fetch(findAccountRecordURL, {
```

Comment:
Does it make sense to make the call here in the `authenticator`? Should we instead do this in the actual application since it seems like the authenticator wouldn't care about fetching the account